sky/cli.py

```diff
@@ -1325,6 +1325,7 @@ def launch(
     In both cases, the commands are run under the task's workdir (if specified)
     and they undergo job queue scheduling.
     """
+    backend_utils.check_network_connection()
```

Comment:
This prints `sky.exceptions.NetworkError: Could not refresh the cluster. Network seems down.`. We're not refreshing the cluster here. We may want to change the message to something else. Ideally we want the message in `check_network_connection` to be something generic, such as `Network seems to be down`, and then higher level methods (like this) can catch this exception and re-raise with appropriate messages (e.g., here `Failed to launch - network seems to be down.`)